src/raps/actions/claimBridge.ts

```diff
@@ -0,0 +1,181 @@
+  }
+
+  let maxBridgeableAmount = sellAmount;
+  let needsNewQuote = false;
+
+  // 1 - Get a quote to bridge the claimed funds
+  const claimBridgeQuote = await getClaimBridgeQuote({
+    chainId,
+    toChainId,
+    fromAddress: address,
+    sellTokenAddress: AddressZero,
+    buyTokenAddress: AddressZero,
+    sellAmount: sellAmount,
+    slippage: 2,
+    swapType: SwapType.crossChain,
+  });
+
+  // if we don't get a quote or there's an error we can't continue
+  if (!claimBridgeQuote || (claimBridgeQuote as QuoteError)?.error) {
+    throw new Error('[CLAIM-BRIDGE]: error getting getClaimBridgeQuote');
+  }
+
+  let bridgeQuote = claimBridgeQuote as CrosschainQuote;
+
+  // 2 - We use the default gas limit (already inflated) from the quote to calculate the aproximate gas fee
+  const initalGasLimit = bridgeQuote.defaultGasLimit as string;
+
+  // Force typing since we only deal with 1559 gas params here
+  const gasParams = parameters.gasParams as TransactionGasParamAmounts;
+  const feeAmount = add(gasParams.maxFeePerGas, gasParams.maxPriorityFeePerGas);
+  const gasFeeInWei = multiply(initalGasLimit, feeAmount);
+
+  // 3 - Check if the user has enough balance to pay the gas fee
+  const provider = getProviderForNetwork(Network.optimism);
+
+  const balance = await provider.getBalance(address);
+
+  // if the balance minus the sell amount is less than the gas fee we need to make adjustments
+  if (lessThan(subtract(balance.toString(), sellAmount), gasFeeInWei)) {
+    // if the balance is less than the gas fee we can't continue
+    if (lessThan(sellAmount, gasFeeInWei)) {
+      throw new Error('[CLAIM-BRIDGE]: error insufficient funds to pay gas fee');
+    } else {
+      // otherwie we bridge the maximum amount we can afford
+      maxBridgeableAmount = subtract(sellAmount, gasFeeInWei);
+      needsNewQuote = true;
+    }
+  }
+
+  // if we need to bridge a different amount we get a new quote
+  if (needsNewQuote) {
+    const newQuote = await getClaimBridgeQuote({
+      chainId,
+      toChainId,
+      fromAddress: address,
+      sellTokenAddress: AddressZero,
+      buyTokenAddress: AddressZero,
+      sellAmount: maxBridgeableAmount,
+      slippage: 2,
+      swapType: SwapType.crossChain,
+    });
+
+    if (!newQuote || (newQuote as QuoteError)?.error) {
+      throw new Error('[CLAIM-BRIDGE]: error getClaimBridgeQuote (new)');
+    }
+
+    bridgeQuote = newQuote as CrosschainQuote;
+  }
+
+  // now that we have a valid quote for the maxBridgeableAmount we can estimate the gas limit
+  let gasLimit;
+  try {
+    gasLimit = await provider.estimateGas({
+      from: address,
+      to: bridgeQuote.to as Address,
+      data: bridgeQuote.data,
+      value: bridgeQuote.value,
+      ...gasParams,
+    });
+  } catch (e) {
+    // Instead of failing we'll try using the default gas limit + 20%
+    gasLimit = (Number(bridgeQuote.defaultGasLimit) * 1.2).toString();
```

Comment:
haven't confirmed but leaving a note: in the error case, will setting the gasLimit like this result in the correct /expected format? could you end up with something with decimals

Reply:
same on bx if so ^